src/pages/Fox/components/FoxFarming.tsx

```diff
@@ -0,0 +1,365 @@
+import { bnOrZero } from 'lib/bignumber/bignumber'
+import { fromBaseUnit } from 'lib/math'
+import { formatSecondsToDuration } from 'lib/utils/time'
+import { marketApi } from 'state/slices/marketDataSlice/marketDataSlice'
+import { foxEthLpAssetId } from 'state/slices/opportunitiesSlice/constants'
+import { opportunitiesApi } from 'state/slices/opportunitiesSlice/opportunitiesApiSlice'
+import { fetchAllOpportunitiesIdsByChainId } from 'state/slices/opportunitiesSlice/thunks'
+import {
+  DefiProvider,
+  DefiType,
+  type LpEarnOpportunityType,
+} from 'state/slices/opportunitiesSlice/types'
+import { serializeUserStakingId, toOpportunityId } from 'state/slices/opportunitiesSlice/utils'
+import {
+  selectAssetById,
+  selectEarnUserStakingOpportunityByOpportunityId,
+  selectMarketDataByAssetIdUserCurrency,
+} from 'state/slices/selectors'
+import { useAppDispatch, useAppSelector } from 'state/store'
+
+import { useFoxPageContext } from '../hooks/useFoxPageContext'
+
+const containerPaddingX = { base: 4, xl: 0 }
+const headerTitleMb = { base: 4, md: 0 }
+const headerTitleMaxWidth = { base: '100%', md: '50%' }
+const headerSx: FlexProps['sx'] = {
+  alignItems: { base: 'flex-start', md: 'center' },
+  justifyContent: 'space-between',
+  mb: 8,
+  flexDir: {
+    base: 'column',
+    md: 'row',
+  },
+}
+const stackProps: StackProps = {
+  width: '100%',
+  flexDir: 'column',
+  flex: 1,
+  spacing: 0,
+}
+const columnsProps = {
+  base: 1,
+  md: 2,
+}
+
+export const FoxFarming = () => {
+  const { assetAccountId } = useFoxPageContext()
+  const translate = useTranslate()
+  const isFoxFarmingEnabled = useFeatureFlag('FoxPageFoxFarmingSection')
+  const history = useHistory()
+  const location = useLocation()
+  const appDispatch = useAppDispatch()
+  const {
+    state: { isConnected, isDemoWallet },
+    dispatch,
+  } = useWallet()
+  const queryClient = useQueryClient()
+
+  useEffect(() => {
+    appDispatch(marketApi.endpoints.findByAssetId.initiate(foxEthLpAssetId))
+  }, [appDispatch])
+
+  const foxEthMarketData = useAppSelector(state =>
+    selectMarketDataByAssetIdUserCurrency(state, foxEthLpAssetId),
+  )
+
+  const opportunityId = useMemo(
+    () =>
+      toOpportunityId({
+        assetNamespace: 'erc20',
+        assetReference: ETH_FOX_STAKING_EVERGREEN_CONTRACT,
+        chainId: assetAccountId ? fromAccountId(assetAccountId).chainId : ethChainId,
+      }),
+    [assetAccountId],
+  )
+
+  const stakingOpportunityByIdFilter = useMemo(() => {
+    const userStakingId = assetAccountId
+      ? serializeUserStakingId(assetAccountId, opportunityId)
+      : undefined
+    return { userStakingId, opportunityId }
+  }, [assetAccountId, opportunityId])
+
+  const opportunity = useAppSelector(state =>
+    selectEarnUserStakingOpportunityByOpportunityId(state, stakingOpportunityByIdFilter),
+  )
+
+  const { getPeriodFinish } = useFoxFarming(ETH_FOX_STAKING_EVERGREEN_CONTRACT)
```

Comment:
nitpick: util exposed but only consumed for query, could expose `periodFinishQuery` instead to make this guy less bigly

Reply:
I mean, every other methods from this contracts are exported as methods, I just followed the exact same pattern, I'm agnostic, changing it in [527f86c](https://github.com/shapeshift/web/pull/7934/commits/527f86ce55071bc81f46808931a053d3e5ee04e3)